hidimstat/Dnn_learner_single.py

```diff
@@ -360,17 +412,21 @@ def __tuning_hyper(self, X, y):
         self.set_params(**best_hyper)
 
     def predict(self, X, scale=True):
-        """Predict regression target for X.
+        """
+        This function predicts the regression target for the input samples X.
+
         Parameters
         ----------
-        X : {array-like, sparse matrix}, shape (n_samples, n_features)
-            The training input samples.
-        scale: bool, default=True
-            The continuous features will be standard scaled or not.
+        X : {array-like, sparse matrix} of shape (n_test_samples, n_features),
+            default=None
+            The input samples.
+        scale : bool, default=True
```

Comment:
Could it be a class parameter rather than an argument to predict ?

Reply:
Which part do you mean?

Reply:
IIRC in sklearn `fit` `predict` and `transform` methods take few and standard argumens: Xtrain, X_test, y_train etc. Can we avoid having other aguments, by converting them to class parameters ?
Is this clear for you ?